hotsos/client.py

```diff
@@ -116,46 +129,41 @@ def minimise(self, summary, mode):
         log.warning("Unknown minimalmode '%s'", mode)
         return summary
 
-    def get(self, fmt='yaml', html_escape=False, minimal_mode=None,
-            plugin=None, max_level=2):
-        if plugin:
-            filtered = {plugin: self._summary[plugin]}
+    def get(self, params: OutputManagerParameters = OutputManagerParameters()):
```

Comment:
Passing mutable types as param defaults is not a good idea imho for the simple reason that changes (however accidental) to the parameter made during one call of get() will persist to future calls to get(). It would be safer to have the caller pass it in rather than set a default.

Reply:
```
from dataclasses import dataclass

@dataclass
class MyOptions:
    flag: bool = False


def foo(flag, opt: MyOptions = MyOptions()):
    if flag:
        opt.flag = True
    else:
        print(opt.flag)

foo(False)
foo(True)
foo(False)
```
 produces

```
foo(False)
foo(True)
foo(False)
```

Reply:
That was an interesting gotcha, thanks Ed!